src_fun/parser.cpp

```diff
@@ -614,6 +625,9 @@ static bool parse_expression_leaf(Token_Stream* stream, Block_Builder* builder,
         if (!take_atom(stream, ATOM_FIRST_IDENTIFIER, "Expected the type alias name after '$'."_s))
             return false;
 
+        if (lookahead_atom(stream, ATOM_COLON, 0))
+            return ReportError(stream->ctx, stream->cursor - 2, "An alias can't have an inferred type because aliases don't have assigned runtime types."_s);

```

Comment:
This is a more general parsing error.
`A: u32 : 1;`
Should not parse for the same reason.

Reply:
Changed it, and added suggestion_remove that adds striketrhough.